holoviews/plotting/bokeh/callbacks.py

```diff
@@ -1173,61 +1217,71 @@ def _watch_position(self):
-                xs = source.get_column(renderer.glyph.x.field)
-                ys = source.get_column(renderer.glyph.y.field)
-              } else if (renderer.glyph.right && renderer.glyph.top) {
-                xs = source.get_column(renderer.glyph.right.field)
-                ys = source.get_column(renderer.glyph.top.field)
-              } else if (renderer.glyph.x1 && renderer.glyph.y1) {
-                xs = source.get_column(renderer.glyph.x1.field)
-                ys = source.get_column(renderer.glyph.y1.field)
-              } else if (renderer.glyph.xs && renderer.glyph.ys) {
-                xs = source.get_column(renderer.glyph.xs.field)
-                ys = source.get_column(renderer.glyph.ys.field)
-              }
-              if (!xs || !ys) { return }
-              for (const i of indices) {
-                let ix = xs[i]
-                let iy = ys[i]
-                let tx, ty
-                if (typeof ix === 'number') {
-                  tx = ix
-                  ty = iy
-                } else {
-                  while (ix.length && (typeof ix[0] !== 'number')) {
-                    ix = ix[0]
-                    iy = iy[0]
-                  }
-                  tx = Math.max(...ix)
-                  ty = Math.max(...iy)
+            export default ({panel, renderer, source, selected, popup_position}, cb_obj, _) => {
+                const el = panel.elements[1];
+                if ((el && !el.visible) || !cb_obj.final) {
+                    return;
                 }
-                if (!x || (tx > x)) {
-                  x = tx
+                let x, y, xs, ys;
+                let indices = selected.indices;
+                if (cb_obj.geometry.type == 'point') {
+                    indices = indices.slice(-1);
                 }
-                if (!y || (ty > y)) {
-                  y = ty
+                if (renderer.glyph.x && renderer.glyph.y) {
+                    xs = source.get_column(renderer.glyph.x.field);
+                    ys = source.get_column(renderer.glyph.y.field);
+                } else if (renderer.glyph.right && renderer.glyph.top) {
+                    xs = source.get_column(renderer.glyph.right.field);
+                    ys = source.get_column(renderer.glyph.top.field);
+                } else if (renderer.glyph.x1 && renderer.glyph.y1) {
+                    xs = source.get_column(renderer.glyph.x1.field);
+                    ys = source.get_column(renderer.glyph.y1.field);
+                } else if (renderer.glyph.xs && renderer.glyph.ys) {
+                    xs = source.get_column(renderer.glyph.xs.field);
+                    ys = source.get_column(renderer.glyph.ys.field);
                 }
-              }
-              if (x && y) {
-                panel.position.setv({x, y})
-              }
-            }""",
+                if (!xs || !ys) { return; }
+
+                let minX = null, maxX = null, minY = null, maxY = null;
```

Comment:
```suggestion
                let minX, maxX, minY, maxY;
```

And change the if statement from `null` to `undefined`